caffe2/core/types.h

```diff
@@ -33,10 +33,10 @@ inline StorageOrder StringToStorageOrder(const string& str) {
 inline constexpr char NameScopeSeparator() { return '/'; }
 
 // From TypeMeta to caffe2::DataType protobuffer enum.
-TensorProto::DataType TypeMetaToDataType(const TypeMeta& meta);
+TensorProto::DataType TypeMetaToDataType(const c10::TypeMeta& meta);
```

Comment:
I think it's actually confusing because have TypeMeta in both places - then why do you invoke it here from c10 namespace?

According to our criteria we discussed:
1) either not move class at all (and use caffe2::TypeMeta directly).
2) move the class proper and update the references throughout